heron/scheduler-core/src/java/org/apache/heron/scheduler/RuntimeManagerRunner.java

```diff
@@ -237,6 +237,46 @@ private int getCurrentContainerNumber(String topologyName) {
     return cPlan.getContainers().size();
   }
 
+  @VisibleForTesting
+  void updateTopologyComponentParallelism(String topologyName, String  newParallelism)
```

Comment:
+1 for Ning. We should reuse the `updatePackingPlan` instead of duplicating code. A special number of `newContainers` like -1 could be provided to indicate there's no container change.